src/main/java/org/folio/rest/impl/PurchaseOrdersAPI.java

```diff
@@ -283,4 +283,43 @@ public void putOrdersStoragePurchaseOrdersById(String id, String lang, org.folio
       }
     });
   }
+
+  private void processCreateSequence(PurchaseOrder entity, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext,
+                                     String tenantId, String persistenceId, OutStream stream) {
+    PostgresClient.getInstance(vertxContext.owner(), tenantId).execute(CREATE_SEQUENCE.getQuery(entity.getId()),
+      createSequenceReply -> {
+        if(createSequenceReply.succeeded()) {
+          Response response = PostOrdersStoragePurchaseOrdersResponse
+            .respond201WithApplicationJson(stream, PostOrdersStoragePurchaseOrdersResponse.headersFor201()
+              .withLocation(PURCHASE_ORDER_LOCATION_PREFIX + persistenceId));
+          respond(asyncResultHandler, response);
+        } else {
+          // TODO Transactionality (???) --> Roll-back PO creation?
+          log.error(createSequenceReply.cause().getMessage(), createSequenceReply.cause());
+          respond(asyncResultHandler, PostOrdersStoragePurchaseOrdersResponse
+            .respond500WithTextPlain(createSequenceReply.cause().getMessage()));
+        }
+      });
+  }
+
+  private void processDropSequence(String lang, PurchaseOrder entity, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext,
+                                   String tenantId, AsyncResult<UpdateResult> reply) {
+    PurchaseOrder.WorkflowStatus status = entity.getWorkflowStatus();
+    if(status == PurchaseOrder.WorkflowStatus.OPEN || status == PurchaseOrder.WorkflowStatus.CLOSED) {
+      PostgresClient.getInstance(vertxContext.owner(), tenantId).select(DROP_SEQUENCE.getQuery(entity.getId()), sequenceDeleteReply -> {
+        if (sequenceDeleteReply.succeeded()) {
+          asyncResultHandler.handle(Future.succeededFuture(PutOrdersStoragePurchaseOrdersByIdResponse
+            .respond204()));
+        } else {
+          // TODO Transactionality (???) --> Roll-back PO update?
```

Comment:
I'd say leave the TODO until we can figure out a better way to handle these failures

Reply:
Please have a look at tips provided by Folijet team on the possible approach to work with transaction: [Chained futures](https://wiki.folio.org/display/FOLIJET/Chained+futures+and+rx-java+overview#Chainedfuturesandrx-javaoverview-Chainedfutures)

Reply:
Ah, thanks for pointing this out to me @piotr-kalashuk